internal/cmd/backup.go

```diff
@@ -118,6 +120,39 @@ func runBackup(c *cli.Context) error {
 		}
 	}
 
+	// Repositories exclude mirror repositories
+	if !c.Bool("exclude-mirror-repos") && !c.Bool("database-only") {
+		reposFile := filepath.Join(rootDir, "non-mirror-repositories.zip")
+		reposInDB, err := db.GetRepositoriesMirror()
```

Comment:
As I mentioned in previous review, we should `for _, repo := range repos` and add Git directory of each repo to a temporary archive, see [`ZipArchive.AddDir`](https://gowalker.org/github.com/Unknwon/cae/zip#ZipArchive_AddDir) method.

Reply:
Roughly looks like this (hand written won't compile):

```go
repoArchive := zip.NewArchive("repositories.zip")
for _, r :=range repos {
	name := filepath.Join(r.OwnerName, r.Name + ".git")
	repoArchive.AddDir(name, filepath.Join(conf.Repository.Root, name)) // don't forget check error
}

Finally you have "repositories.zip", then add this to the bigger ZIP archive as you do right now.
```